allauth/socialaccount/providers/__init__.py

```diff
@@ -15,6 +15,23 @@ def get_list(self, request=None):
             provider_cls(request)
             for provider_cls in self.provider_map.values()]
 
+    def get_site_list(self, request=None, site=None):
+        from allauth.socialaccount.models import SocialApp
+
+        self.load()
+        providers = []
+
+        for provider_cls in self.provider_map.values():
+            provider = provider_cls(request)
+
+            active_sites = SocialApp.objects.filter(
```

Comment:
@BradLucky is there a way to reverse this query and get it out of the `for` loop? For instance, instead of: 

```
for each provider...
    get a list of all of the sites...
    check if the site is in the provider...
    repeat...
```

Something like this:

```
Get a list of all of the providers that the site belongs to...

for each provider...
    check if the site is in the provider...
    repeat...
```

Reply:
Good call. Surely that can be done. Doing it inside the loop was a carry-over from a different approach I was taking. Thanks for noticing it!

Reply:
Fixed.